core/corehttp/gateway_handler.go

```diff
@@ -324,61 +344,34 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 	logger := log.With("from", r.RequestURI)
 	logger.Debug("http request received")
 
-	// X-Ipfs-Gateway-Prefix was removed (https://github.com/ipfs/go-ipfs/issues/7702)
-	// TODO: remove this after  go-ipfs 0.13 ships
-	if prfx := r.Header.Get("X-Ipfs-Gateway-Prefix"); prfx != "" {
-		err := fmt.Errorf("X-Ipfs-Gateway-Prefix support was removed: https://github.com/ipfs/go-ipfs/issues/7702")
-		webError(w, "unsupported HTTP header", err, http.StatusBadRequest)
+	var (
+		err          error
+		shouldReturn bool
```

Comment:
`shouldReturn` is very undescriptive. It tells *what* you are doing. However I already know *what* you are doing just be reading the code.
```go
if shouldReturn {
  return
}
```
Is good enough.

Here I would name it `requestHandled` instead, as it tells *why* you are doing it.

Reply:
I chose `shouldReturn` because it indicates if processing of the request should halt.  Maybe `shouldHalt` would be better?  It isn't as simple as saying the request was handled, because some handling requires halting request processing and some does not.  Does that make sense?

Reply:
I think `shouldHalt` is failing for the same reasons.

First as of [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119) should means recommended but optional.
While this code would be wrong if you didn't returned, so it's a `MUST` not `SHOULD`.
I know 2119 only applies to `SHOULD` and not `should`, but still I think it is confusing.

Secondly `shouldHalt` explain *what* not *why*.
The *what* is already self explainatory, I find `if a { return }` to be perfectly clear, so if the variable could explain instead *why* instead of *what* the code would be clearer because instead of making clearer something that is bright white, it would make clearer something that is unclear.